projects/aca-content/src/lib/components/search/search-input/search-input.component.spec.ts

```diff
@@ -1,62 +1,116 @@
+    component.searchInputControl.searchFieldFormControl.setValue('');
+    component.searchInputControl.searchFieldFormControl.markAsTouched();
+    fixture.detectChanges();
+    expect(getFirstError()).toBe('SEARCH.INPUT.REQUIRED');
   });
 
-  describe('Validation Behavior', () => {
-    function getFirstError(): string {
-      const error = fixture.debugElement.query(By.directive(MatError));
-      return error?.nativeElement.textContent.trim();
-    }
-
-    it('should show required error when field is empty and touched', () => {
-      component.searchInputControl.searchFieldFormControl.setValue('');
-      component.searchInputControl.searchFieldFormControl.markAsTouched();
-      fixture.detectChanges();
-
-      const error = getFirstError();
-      fixture.detectChanges();
-      expect(error).toBe('SEARCH.INPUT.REQUIRED');
-    });
-
-    // it('should not show error when field has value', () => {
-    //   mockControl.searchFieldFormControl.setValue('test');
-    //   mockControl.searchFieldFormControl.markAsTouched();
-    //   fixture.detectChanges();
-
-    //   const error = fixture.debugElement.query(By.directive(MatError));
-    //   expect(error).toBeNull();
-    // });
-
-    // it('should not show error when field is untouched', () => {
-    //   mockControl.searchFieldFormControl.setValue('');
-    //   mockControl.searchFieldFormControl.markAsUntouched();
-    //   fixture.detectChanges();
-
-    //   const error = fixture.debugElement.query(By.directive(MatError));
-    //   expect(error).toBeNull();
-    // });
+  it('should not show error when field has value', () => {
+    openSearchContainer();
+    component.searchInputControl.searchFieldFormControl.setValue('test');
+    component.searchInputControl.searchFieldFormControl.markAsTouched();
+    fixture.detectChanges();
+    const error = fixture.debugElement.query(By.directive(MatError));
+    expect(error).toBeNull();
+  });
+
+  it('should not show error when field is untouched', () => {
+    openSearchContainer();
+    component.searchInputControl.searchFieldFormControl.setValue('');
+    component.searchInputControl.searchFieldFormControl.markAsUntouched();
+    fixture.detectChanges();
+    const error = fixture.debugElement.query(By.directive(MatError));
+    expect(error).toBeNull();
+  });
+
+  it('should dispatch SearchByTermAction when libraries are checked and term is new', () => {
+    spyOn(component as any, 'isLibrariesChecked').and.returnValue(true);
+    spyOn(component as any, 'isFoldersChecked').and.returnValue(false);
+    spyOn(component as any, 'isFilesChecked').and.returnValue(false);
+    component.searchedWord = 'test';
+    component.onSearchSubmit('Enter');
+    expect(store.dispatch).toHaveBeenCalled();
+  });
+
+  it('should not dispatch SearchByTermAction when no checkboxes are selected and term is empty', () => {
+    store.dispatch.calls.reset();
+
+    spyOn(component as any, 'isLibrariesChecked').and.returnValue(false);
```

Comment:
the same like https://github.com/Alfresco/alfresco-content-app/pull/4683/commits/b587f5d1bf7509cd9f1a94485013a6ba8945a098#r2221681874